core/identity-management/iin-agent/src/common/utils.ts

```diff
@@ -14,3 +30,238 @@ export function handlePromise<T>(promise: Promise<T>): Promise<[T?, Error?]> {
+        if (ledgerBase.orgMspId != memberId) {
+            throw new Error(`This IIN Agent's member Id: ${ledgerBase.orgMspId} doesn't match with provided member Id: ${memberId} in request.`);
+        }
+        return ledgerBase;
+    } else {
+        throw new Error(`DLT Type ${process.env.DLT_TYPE} not implemented`);
+    }
+}
+
+export function getLedgerId(securityDomain: string): any {
+    const secDomConfigPath = process.env.SECURITY_DOMAIN_CONFIG_PATH ? process.env.SECURITY_DOMAIN_CONFIG_PATH : path.resolve(__dirname, "../", "../", "security-domain-config.json");
+    if (!fs.existsSync(secDomConfigPath)) {
+        throw new Error('Security Domain config does not exist at path: ' + secDomConfigPath);
+    }
+    const secDomConfig = JSON.parse(fs.readFileSync(secDomConfigPath, 'utf8').toString());
+    return secDomConfig[securityDomain];
+}
+
+export function getSecurityDomainDNS(securityDomain: string): any {
+    const dnsConfigPath = process.env.DNS_CONFIG_PATH ? process.env.DNS_CONFIG_PATH : path.resolve(__dirname, "../", "../", "dnsconfig.json");
+    if (!fs.existsSync(dnsConfigPath)) {
+        throw new Error('DNS config does not exist at path: ' + dnsConfigPath);
+    }
+    const dnsConfig = JSON.parse(fs.readFileSync(dnsConfigPath, 'utf8').toString());
+    return dnsConfig[securityDomain];
+}
+
+export function getAllSecurityDomainDNS(): any {
+    const dnsConfigPath = process.env.DNS_CONFIG_PATH ? process.env.DNS_CONFIG_PATH : path.resolve(__dirname, "../", "../", "dnsconfig.json");
+    if (!fs.existsSync(dnsConfigPath)) {
+        throw new Error('DNS config does not exist at path: ' + dnsConfigPath);
+    }
+    const dnsConfig = JSON.parse(fs.readFileSync(dnsConfigPath, 'utf8').toString());
+    return dnsConfig;
+}
+
+
+
+export function defaultCallback(err: any, response: any) {
+    if (response) {
+        console.log(`IIN Agent Response: ${JSON.stringify(response.toObject())}`);
+    } else if (err) {
+        console.log(`IIN Agent Error: ${JSON.stringify(err)}`);
+    }
+}
+
+export async function delay(ms: number) {
+    await new Promise(f => setTimeout(f, ms));
+}
+
+/**
+ * Sign a message using SHA256
+ * message: string
+ * privateKey: pem string
+ * returns: signature in base64 string
+**/
+export function signMessage(message, privateKey) {
+    const sign = crypto.createSign("SHA256");
+    sign.write(message);
+    sign.end();
+    return sign.sign(privateKey).toString('base64');
+};
+/**
+ * Verifies a signature over message using SHA256
+ * message: string
+ * certificate: pem string
+ * signature: base64 string
+ * returns: True/False
+ **/
+export function verifySignature(message, certificate, signature) {
+    const messageBuffer = Buffer.from(message);
+    const signBuffer = Buffer.from(signature, 'base64');
+    const publicKey = crypto.createPublicKey(certificate).export({type:'spki', format:'pem'});
```

Comment:
Does the "type" matter here?

Reply:
From docs it needs to be provided for public keys, quoting from docs
`Must be one of 'pkcs1' (RSA only) or 'spki'.`
I'm not sure what each of those mean, I tried `spki` and it was working.